ext/php8/ddtrace.c

```diff
@@ -80,17 +84,35 @@ static int ddtrace_startup(struct _zend_extension *extension) {
     // This touches global state, which, while unlikely, may play badly when interacting with other extensions, if done
     // post-startup
     ddtrace_internal_handlers_startup();
+
+#ifdef PHP_DATADOG_PROFILING_H
+    return datadog_profiling_startup(extension);
+#else
     return SUCCESS;
+#endif
 }
 
 static void ddtrace_shutdown(struct _zend_extension *extension) {
+    ddtrace_internal_handlers_shutdown();
+
+#ifdef PHP_DATADOG_PROFILING_H
+    datadog_profiling_shutdown(extension);
+#else
     UNUSED(extension);
+#endif
```

Comment:
Since the profiler was started up after the internal handlers, would it make sense to shut the profiler down before the handlers? Basically it's the last to start up so I would expect it to be the first to shut down.